pysam/libcbcf.pyi

```diff
@@ -0,0 +1,457 @@
+    @overload
+    def add_meta(self, key: str, value: str = ..., items: None = ...) -> None: ...
+    def add_sample(self, name: str) -> None: ...
+
+class VariantRecordFilter:
+    def __len__(self) -> int: ...
+    def __bool__(self) -> bool: ...
+    def __getitem__(self, key: Union[str, int]) -> VariantMetadata: ...
+    def add(self, key: str) -> None: ...
+    def __delitem__(self, key: Union[str, int]) -> None: ...
+    def clear(self) -> None: ...
+    def __iter__(self) -> Iterator[str]: ...
+    def get(
+        self, key: Union[str, int], default: _D = ...
+    ) -> Union[_D, VariantMetadata]: ...
+    def __contains__(self, key: str) -> bool: ...
+    def iterkeys(self) -> Iterator[str]: ...
+    def itervalues(self) -> Iterator[VariantMetadata]: ...
+    def iteritems(self) -> Iterator[Tuple[str, VariantMetadata]]: ...
+    def keys(self) -> List[str]: ...
+    def items(self) -> List[Tuple[str, VariantMetadata]]: ...
+    def values(self) -> List[VariantMetadata]: ...
+    def __eq__(self, other) -> bool: ...
+    def __ne__(self, other) -> bool: ...
+
+class VariantRecordFormat:
+    def __len__(self) -> int: ...
+    def __bool__(self) -> bool: ...
+    def __getitem__(self, key: str) -> VariantMetadata: ...
+    def __delitem__(self, key: str) -> None: ...
+    def clear(self) -> None: ...
+    def __iter__(self) -> Iterator[str]: ...
+    def get(self, key: str, default: _D = ...) -> Union[_D, VariantMetadata]: ...
+    def __contains__(self, other) -> bool: ...
+    def iterkeys(self) -> Iterator[str]: ...
+    def itervalues(self) -> Iterator[VariantMetadata]: ...
+    def iteritems(self) -> Iterator[Tuple[str, VariantMetadata]]: ...
+    def keys(self) -> List[str]: ...
+    def items(self) -> List[Tuple[str, VariantMetadata]]: ...
+    def values(self) -> List[VariantMetadata]: ...
+
+_InfoValue = Any  # TODO see bcf_info_get_value
+
+class VariantRecordInfo:
+    def __len__(self) -> int: ...
+    def __bool__(self) -> bool: ...
+    def __getitem__(self, key: str) -> _InfoValue: ...
+    def __setitem__(self, key: str, object: _InfoValue) -> None: ...
+    def __delitem__(self, key: str) -> None: ...
+    def clear(self) -> None: ...
+    def __iter__(self) -> Iterator[str]: ...
+    def get(self, key: str, default: _D = ...) -> Union[_D, _InfoValue]: ...
+    def __contains__(self, key: str) -> bool: ...
+    def iterkeys(self) -> Iterator[str]: ...
+    def itervalues(self) -> Iterator[_InfoValue]: ...
+    def iteritems(self) -> Iterator[Tuple[str, _InfoValue]]: ...
+    def keys(self) -> List[str]: ...
+    def items(self) -> List[Tuple[str, _InfoValue]]: ...
+    def values(self) -> List[_InfoValue]: ...
+    def update(
+        self, items: Optional[Mapping[str, _InfoValue]] = ..., **kwargs
+    ) -> None: ...
+    def pop(self, key: str, default: _D = ...) -> Union[_D, _InfoValue]: ...
+    def __eq__(self, other) -> bool: ...
+    def __ne__(self, other) -> bool: ...
+
+class VariantRecordSamples:
```

Comment:
Can be indexed by index and sample name, so Mapping[int | str, VariantRecordSample]?